app/lib/frontend/handlers/admin.dart

```diff
@@ -8,13 +8,29 @@ import 'dart:async';
 
 import 'package:shelf/shelf.dart' as shelf;
 
+import '../../account/backend.dart';
 import '../../shared/handlers.dart';
 
 import '../backend.dart';
 import '../models.dart';
 import '../templates/admin.dart';
 import '../templates/misc.dart';
 
+/// Handles requests for /oauth/callback
+shelf.Response oauthCallbackHandler(shelf.Request request) {
+  final code = request.requestedUri.queryParameters['code'];
+  final state = request.requestedUri.queryParameters['state'];
+  if (code == null || state == null) {
+    return notFoundHandler(request);
+  }
+  return redirectResponse(request.requestedUri
+      .replace(
+        path: state,
+        queryParameters: request.requestedUri.queryParameters,
```

Comment:
This is an open redirect vulnerability...

Let's just send a state as follows: `{action: 'accept-invite', package: '...', ...}`